sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryElement.Serialization.cs

```diff
@@ -140,11 +140,96 @@ private void SerializeLiteral(Utf8JsonWriter writer)
                     }
                     break;
                 default:
-                    writer.WriteObjectValue(Literal!);
+                    WriteObjectValue(writer, Literal!, options);
                     break;
             }
         }
 
+        private void WriteObjectValue<T1>(Utf8JsonWriter writer, T1 value, ModelReaderWriterOptions options)
```

Comment:
Why repeat all of this base logic here rather than delegating to the exiting method if you're not dealing with an `IJsonModel`?

Reply:
We can't leverage the existing [Utf8JsonWriterExtensions ](https://github.com/Azure/azure-sdk-for-net/blob/ceee0af493744f1dbcb40aa0a3274f761f10018f/sdk/core/Azure.Core/src/Shared/Utf8JsonWriterExtensions.cs#L71) method.
 Because if we do so, and it falls into the KeyValuePair and Array case [here](https://github.com/Azure/azure-sdk-for-net/blob/ceee0af493744f1dbcb40aa0a3274f761f10018f/sdk/core/Azure.Core/src/Shared/Utf8JsonWriterExtensions.cs#L127-L143), it goes into the recursion of the existing method, which is missing IJsonModel handling and it will just throw.

That's why I think the existing method should be removed if we can handle it internally in DFE, and the existing method is not used in Azure.Core code except tests.

Reply:
The best I can do is to use the existing method for the primitive types handling.
Please let me know if you think this should be further refined.

Reply:
I will not remove it in this PR due to the lack of knowledge in testing of Azure.Core.

Reply:
Created https://github.com/Azure/azure-sdk-for-net/issues/50508 to follow up the cleanup